scripts/generate-components-docs.rb

```diff
@@ -786,21 +1000,73 @@ def apply_schema_metadata!(source_schema, resolved_schema)
   # of optional key/value pairs i.e. tags -- and it allows templateable values.
+
+  # Handle adding any defined examples to the type definition.
+  #
+  # TODO: Since a resolved schema could be represented by _multiple_ input types, and we only take
+  # examples in the form of strings, we don't have a great way to discern which type definition
+  # should get the examples applied to it (i.e. for a number-or-enum-of-strings schema, do we
+  # apply the examples to the number type def or the enum type def?) so we simply... apply them to
+  # any type definition present in the resolved schema.
+  #
+  # We might be able to improve this in the future with a simply better heuristic, dunno. This might
+  # also work totally fine as-is!
+  examples = get_schema_metadata(source_schema, 'docs::examples')
+  if !examples.nil?
+    count = [examples].flatten.length
+    @logger.debug "Schema has #{count} example(s)."
+    resolved_schema['type'].each { |type_name, type_def|
+      # We need to recurse one more level if we're dealing with an array type definition, as we need
+      # to stick the examples on the type definition for the array's `items`. There might also be
+      # multiple type definitions under `items`, but we'll cross that bridge if/when we get to it.
+      case type_name
+      when 'array'
+        type_def['items']['type'].each { |subtype_name, subtype_def|
+          if subtype_name != 'array'
+            subtype_def['examples'] = [examples].flatten
+          end
+        }
+      else
+        type_def['examples'] = [examples].flatten
+      end
+    }
+  end
+
+  # Apply any units that have been specified.
+  type_unit = get_schema_metadata(source_schema, 'docs::type_unit')
+  if !type_unit.nil?
+    schema_type = numeric_schema_type(resolved_schema)
+    if !schema_type.nil?
+      resolved_schema['type'][schema_type]['unit'] = type_unit.to_s
+    end
+  end
+
+  # Modify the `syntax` of a string type definition if overridden via metadata.
+  syntax_override = get_schema_metadata(source_schema, 'docs::syntax_override')
+  if !syntax_override.nil?
+    if resolved_schema_type?(resolved_schema) != "string"
+      @logger.error "Non-string schemas should not use the `syntax_override` metadata attribute."
+      exit
+    end
+
+    resolved_schema['type']['string']['syntax'] = syntax_override.to_s
+  end
 end
 
 def get_rendered_description_from_schema(schema)
   # If the schema is marked as `no_description`, we're being told -- for whatever reason -- that the
   # existing title/description should not be rendered in the output. This is primarily to avoid
   # spitting out developer-oriented documentation into the user-facing documentation, when we're
   # providing the necessary description in another way.
-  if !get_schema_metadata(schema, 'docs::no_description').nil?
-    return ''
-  end
+  #if !get_schema_metadata(schema, 'docs::no_description').nil?
+  #  return ''
+  #end
```

Comment:
Intended to be be left in?

Reply:
Hmm, I think it can be removed. I was trying to deal with some areas where descriptions were being combined -- like the description on a field of `T` being combined with the description for `T` itself -- but I think I resolved it another way.

I'll take another look and figure it out.

Reply:
This ended up not being needed anymore, so I removed it via dc518595.